modules/k8s/minikube.go

```diff
@@ -1,15 +1,36 @@
 package k8s
 
-import "github.com/gruntwork-io/terratest/modules/testing"
+import (
+	"strings"
+
+	"github.com/gruntwork-io/terratest/modules/testing"
+)
 
 // IsMinikubeE returns true if the underlying kubernetes cluster is Minikube. This is determined by getting the
 // associated nodes and checking if:
 // - there is only one node
-// - the node is named "minikube"
+// - the node has at least one label namespaced with "minikube.k8s.io"
 func IsMinikubeE(t testing.TestingT, options *KubectlOptions) (bool, error) {
 	nodes, err := GetNodesE(t, options)
 	if err != nil {
 		return false, err
 	}
-	return len(nodes) == 1 && nodes[0].GetName() == "minikube", nil
+
+	// ASSUMPTION: Minikube always only has one node.
```

Comment:
Is that a necessary assumption to make? Or is it safer to loop over all nodes and all labels and see if any of them has the minikube label?

Reply:
> Or is it safer to loop over all nodes and all labels and see if any of them has the minikube label?

The safest would be to validate that all nodes have the minikube label. If any node does not have the minikube label, it will most likely be in a mixed cluster mode with minikube and non-minikube nodes, so we should assume it is not minikube.

Reply:
Updated: d05f5a1